test/parallel/test-util-text-encoder.js

```diff
@@ -0,0 +1,50 @@
+'use strict';
+
+require('../common');
+const assert = require('assert');
+const { TextEncoder, TextDecoder } = require('util');
+
+const encoder = new TextEncoder();
+const decoder = new TextDecoder();
+
+assert.strictEqual(decoder.decode(encoder.encode('')), '');
+assert.strictEqual(decoder.decode(encoder.encode('latin1')), 'latin1');
+assert.strictEqual(decoder.decode(encoder.encode('Yağız')), 'Yağız');
+
+// For loop is required to trigger the fast path for encodeInto
+// Since v8 fast path is only triggered when v8 optimization starts.
```

Comment:
There is no guarantee that V8 would take the fast path.

```suggestion
// Using a loop increases the chances of triggering the fast path for encodeInto
// because V8 heuristically optimizes based on information gathered at runtime.
```